File: week4/NoizBuster/3d.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the surfaceArea function below.
def surfaceArea(A):
    # IV = roof and floor (horizental)
    price = H * W * 2

    # vertical surface
    for i in range(H):
        for j in range(W):
            # border
            price += A[i][j] if i == 0 else 0
            price += A[i][j] if j == 0 else 0
            price += A[i][j] if i == H-1 else 0
            price += A[i][j] if j == W-1 else 0
            # in middle
            price += abs(A[i][j] - A[i][j-1]) if j != 0 else 0
            price += abs(A[i][j] - A[i-1][j]) if i != 0 else 0
    return price

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    HW = input().split()

    H = int(HW[0])

    W = int(HW[1])

    A = []

    for _ in range(H):
        A.append(list(map(int, input().rstrip().split())))

    result = surfaceArea(A)

    fptr.write(str(result) + '\n')

    fptr.close()

File: season001/20180917-week4/MonkeyThunder/3D.cpp
int surfaceArea(vector<vector<int>> A) {

    int INT_Result;
    int INT_BuffBlockNum;

    INT_Result = 0;

    for (int i0 = 0; i0 < A.size(); i0++) {
        for (int i1 = 0; i1 < A[i0].size(); i1++) {
            //가로, 세로 블럭 갯수 차이 = 표먼적

            //가로
            if (i0 == 0) {
                INT_BuffBlockNum = 0;//모서리
            } else {
                INT_BuffBlockNum = A[i0 - 1][i1];
            }

            INT_Result = INT_Result + abs(A[i0][i1] - INT_BuffBlockNum);

            //세로
            if (i1 == 0) {
                INT_BuffBlockNum = 0;//모서리
            } else {
                INT_BuffBlockNum = A[i0][i1 - 1];
            }

            INT_Result = INT_Result + abs(A[i0][i1] - INT_BuffBlockNum);

            //모서리
            if (i0 == A.size() - 1) {
                INT_Result = INT_Result + A[i0][i1];
            }
            if (i1 == A[i0].size() - 1) {
                INT_Result = INT_Result + A[i0][i1];
            }
        }

    }

    //윗면 or 아랫면 갯수

    INT_BuffBlockNum = 0;
    for (int i0 = 0; i0 < A.size(); i0++) {
        for (int i1 = 0; i1 < A[i0].size(); i1++) {
            INT_BuffBlockNum++;
        }
    }
    //총합
    INT_Result = INT_Result + INT_BuffBlockNum + INT_BuffBlockNum;

    return INT_Result;
}